src/Data/Nat/Binary/Base.agda

```diff
@@ -106,6 +113,38 @@ _        * zero     =  zero
 1+[2 x ] * 2[1+ y ] =  2[1+ y + x * 2[1+ y ] ]
 1+[2 x ] * 1+[2 y ] =  1+[2 x + y * 1+[2 x ] ]
 
+_-_ : ℕᵇ → ℕᵇ → Maybe ℕᵇ
+x        - zero     = just x
+zero     - _        = nothing
+2[1+ x ] - 2[1+ y ] = map double (x - y)
+2[1+ x ] - 1+[2 y ] = map 1+[2_] (x - y)
+1+[2 x ] - 2[1+ y ] = map 1+[2_] (x - y >>= predᴹ)
+1+[2 x ] - 1+[2 y ] = map double (x - y)
+
+_divMod_ : ℕᵇ → ℕᵇ → ℕᵇ × ℕᵇ
+m divMod n = maybe′ (flip′ helper (pred n)) (zero , zero) (predᴹ m)
+  module divMod where
+  -- helper m n computes (suc m / suc n, suc m % suc n)
+  helper : ℕᵇ → ℕᵇ → ℕᵇ × ℕᵇ
+  helper zero     zero = 1+[2 zero ] , zero
+  helper zero     _    = zero        , 1+[2 zero ]
+  helper 2[1+ m ] n    = let q , r = helper m n
+                             q₂    = double q
```

Comment:
Can we rename these `2q` instead of `q₂`. A little bit more meaningful.

Reply:
I also renamed `r₂` to `2r` in my latest commit, in line with your suggestion.